docs/enhancements/hypershift-adaptation.md

```diff
@@ -0,0 +1,99 @@
+
+## Proposal
+
+### Performance Profile
+
+The proposal to handle the Performance Profile (PP) mimics the way NTO handles it `Tuned` API
+
+- New `.spec.performanceProfileConfig` will be created in `NodePool` CRD
+- [Cluster Service Consumer](https://hypershift-docs.netlify.app/reference/concepts-and-personas/) will create `PerformanceProfile` objects inside of configmaps, then reference these configmaps objects in the `.spec.performanceProfileConfig` field of the `NodePool` API
+- NodePool Controller will read these configmaps, extract the `PerformanceProfile`s and propagate that information into one `configmap` per `NodePool` in the `hosted-control-plane-namespace` where Performance Profile Controller runs and can read them.
+  - The `configmap` created by NodePool Controller will have the following labels and annotations:
+    - label: `hypershift.openshift.io/performanceprofile-config` : `true`
+    - label: `hypershift.openshift.io/nodePool` : `NodePool` API name where it was referenced.
+    - annotation: `hypershift.openshift.io/nodePool` : `NodePool` API name where it was referenced.
+- Performance Profile Controller will watch these configmaps and extract the `PerformanceProfile` objects from them to generate its output.
+
+### Tuned
+
+The proposal for this output object is to use the way NTO has already put in place to handle its own `tuned` objects as it is.
+
+- Once Performance Profile Controller has created the `tuned` object as usual, it will embeded the `tuned` into a `configmap` in the `hosted-control-plane-namespace`.
+  - This `configmap` will have:
+    - label: `hypershift.openshift.io/tuned-config` : `true`
+    - annotation: `hypershift.openshift.io/nodePool` : `NodePool` API name where the `PeformanceProfile` which generate this `tuned` was referenced.
+- This will trigger the reconcile operation in NTO for these kind of objects.
+
+### MachineConfig
+
+The proposal for this output object is to use the way NTO has already put in place to handle its own `MachineConfig` objects as it is.
+
+- Once Performance Profile Controller has created the `MachineConfig` object as usual, it will embeded the object into a `configmap` in the `hosted-control-plane-namespace`.
+- This `configmap` will have:
+  - label: `hypershift.openshift.io/nto-generated-machine-config` : `true`
+  - label: `hypershift.openshift.io/nodePool` : `NodePool` API name where the `PeformanceProfile` which generate this `MachineConfig` was referenced.
+- This will trigger the reconcile operation in NodePool Controller for these objects.
+
+### MCO KubeletConfig
+
+__WIP__
+
+Being this object also handled by MachineConfig Operator (MCO) as `MachineConfig`s the proposal is to handle them in a similar way.
+
+- Once Performance Profile Controller has created the `Kubeletconfig` object as usual, it will embeded the object into a `configmap` in the `hosted-control-plane-namespace`.
```

Comment:
I haven't tried out creating KubeletConfigs yet on HyperShift. Unless I'm missing something, [they will not be reconciled and applied](https://github.com/openshift/hypershift/blob/338770449bd0b6e14df123d2c17846c774446a63/hypershift-operator/controllers/nodepool/nodepool_controller.go#L1448). Some experimentation / discussion with the HyperShift folks is probably necessary here.

Reply:
On the other hand maybe talking with Hypershift people would help to dissipate any doubt.

Reply:
I think you're right, I had misunderstood `getConfig`, and was thinking the manifest would be ignored if it were a KubeletConfig. It looks like a KubeletConfig is added to the token secret just like MachineConfigs, and should (somehow) be applied.